src/v2/Apps/ViewingRoom/__tests__/ViewingRoomApp.jest.tsx

```diff
@@ -324,13 +325,19 @@ describe("ViewingRoomApp", () => {
       expect(wrapper.find("ViewingRoomTabBar").length).toBe(0)
       expect(wrapper.find("ViewingRoomContentNotAccessible").length).toBe(0)
       expect(wrapper.html()).not.toContain("some child")
-
-      expect(mediator.trigger).toBeCalledWith("open:auth", {
-        mode: "signup",
-        redirectTo: "http://localhost/" + slug,
-        contextModule: "viewingRoom",
-        intent: "viewViewingRoom",
-      })
+      expect(mediator.trigger).toBeCalled()
+    })
+    xit("redirects user to /viewing-rooms on close", () => {
+      // const spy = jest.fn()
+      // ;(useRouter as jest.Mock).mockImplementation(() => {
+      //   return {
+      //     router: {
+      //       push: spy,
+      //     },
+      //   }
+      // })
+      // expect(spy).toHaveBeenCalled()
```

Comment:
@damassi just to update you on the progress of this test, I was able to get the mock implementation set up like so:

```
const spy = jest.fn()
const mockRouter = useRouter as jest.Mock
mockRouter().router.push.mockImplementation(spy)
const wrapper = await getWrapper()
```

However, I can't find the modal in `wrapper` to try to close it. 

I know that the modal opens because I can see a `console.log` in [`ModalManager.openModal`](https://github.com/artsy/force/blob/e517f6ba80480089826e93efa10f6dcb2374ab6e/src/v2/Components/Authentication/Desktop/ModalManager.tsx#L57-L67), but it doesn't seem to be getting rendered in this context.

I'll keep working on this tomorrow, but I am leaning towards merging this work and continuing the test in another PR.

Reply:
@eessex - might have more insight here. I _thought_ that the modal was attached to the dom via a `React.portal` which connects to [#react-portal](https://github.com/artsy/force/blob/6f8ce70c3acf8a75964c0afcf5c96f02aa0e61ab/src/desktop/components/main_layout/templates/artsy_v2.jade#L19), but digging around, I actually can't figure out how the auth component gets rendered at the bottom-most layer. 

In any case, I think your hunch about context is correct.

Reply:
(As a side note, we've got `ModalManager`, `ModalWrapper`, and `Modal` classes all interwoven with one another. I think there's room to collapse these things into a single class / file and save a lot of future dev confusion as they're each roughly equal conceptually.)